### PR TITLE
[Vidbit] Add new extractor (Closes #9688)

### LINKED ISSUE #9688: Support for www.vidbit.co
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.06.03*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.06.03**

### Before submitting an *issue* make sure you have:
- [ ] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [x] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):
```
$ youtube-dl -v <your command line>
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'http://www.youtube.com/watch?v=BaW_jenozKcj']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2016.06.03
[debug] Python version 2.7.11 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: ffmpeg N-75573-g1d0487f, ffprobe N-75573-g1d0487f, rtmpdump 2.4
[debug] Proxy map: {}
...
<end of log>
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: https://www.youtube.com/watch?v=BaW_jenozKc
- Single video: https://youtu.be/BaW_jenozKc
- Playlist: https://www.youtube.com/playlist?list=PL4lCao7KL_QFVb7Iudeipvc2BCavECqzc

---

Hi, seeing as ZippCast is shutting down on June 5th, people are trying to find alternatives. In one of the newest ZippCast videos, this YouTube clone was mentioned. URL: http://www.vidbit.co/ 

Here's an example of a video: http://www.vidbit.co/watch?v=WxAZRmMMgM

Hoping for support for this site and other related 2000s YouTube-like site.

### PR DESCRIPTION
**Pull request by TRox1972**

### CODE REVIEW

File: youtube_dl/extractor/vidbit.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    _BASE_URL = 'http://vidbit.co/%s'
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        return {
+            'id': video_id,
+            'title': self._html_search_regex(r'<h1>(.+)</h1>', webpage, 'title'),
+            'url': self._BASE_URL % self._html_search_regex(r'file:\s*["\'](.+)["\']', webpage, 'video URL'),
```
Comment (dstftw) [inline youtube_dl/extractor/vidbit.py]: `url` should be used as base URL.

Comment (dstftw) [inline youtube_dl/extractor/vidbit.py]: `["\'](.+)["\']` will capture everything including `"` and `'` till trailing `"` or `'`.

File: youtube_dl/extractor/vidbit.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            'thumbnail': self._BASE_URL % self._html_search_regex(r'image:\s*["\'](.*)["\']', webpage, 'thumbnail', fatal=False),
```
Comment (dstftw) [inline youtube_dl/extractor/vidbit.py]: This breaks when there is no thumbnail.

File: youtube_dl/extractor/vidbit.py (hunk 1 above)
Comment (TRox1972) [inline youtube_dl/extractor/vidbit.py]: It seems like the site both uses single and double quotes. Changing it to e.g. `([^"\'])` wouldn't extract the correct title if it contains one type of quotes. Do you know of a good regex for this?

Comment (dstftw) [inline youtube_dl/extractor/vidbit.py]: Capture opening quote and match it with closing.

**Comment by yan12125** [comment]
This is a custom JWPlayer. Should use ```JWPlatformBaseIE._parse_jwplayer_data()``` instead.

**Comment by dstftw** [comment]
Current `_parse_jwplayer_data` won't work with this input since the path is relative.

**Comment by yan12125** [comment]
```_parse_jwplayer_data``` can be extended to support relative URLs.

**Comment by TRox1972** [comment]
So should `_parse_jwplayer_data`be modified to handle relative URLs, or should I use current approach?

**Comment by yan12125** [comment]
It's OK to just use the current approach. Rewriting jwplayer-related codes is not a top priority.

File: youtube_dl/extractor/vidbit.py (hunk 1 above)
Comment (TRox1972) [inline youtube_dl/extractor/vidbit.py]: > `url` should be used as base URL.

@dsftw I'm not sure what you mean. I've seen the same setup in other extractor.

Comment (dstftw) [inline youtube_dl/extractor/vidbit.py]: I mean exactly that: base part of `url` should be used as base URL instead of this hardcode.

**Comment by TRox1972** [comment]
I've added a quick fix for the base URL, but it's not very pretty, so any suggestions for a better solution are appreciated :)

**Comment by dstftw** [comment]
Use `urljoin`?

**Comment by TRox1972** [comment]
@dfstw  Does changes seem OK?

File: youtube_dl/extractor/vidbit.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        return {
+            'id': video_id,
+            'title': self._html_search_regex(r'<h1>(.+)</h1>', webpage, 'title'),
+            'url': compat_urlparse.urljoin(url, self._html_search_regex(r'file:\s*(["\'])((?:(?!\1).)+)\1', webpage, 'video URL', group=2)),
+            'thumbnail': compat_urlparse.urljoin(url, self._html_search_regex(r'image:\s*(["\'])((?:(?!\1).)+)\1', webpage, 'thumbnail', None, group=2)),
```
Comment (dstftw) [inline youtube_dl/extractor/vidbit.py]: This will fail no thumbnails extracted.

**Comment by dstftw** [comment]
Also carry long like and squash commits.

**Comment by TRox1972** [comment]
@dstftw Does this seem good?

**Pull request closed without merging**